packages/flutter_test/lib/src/controller.dart

```diff
@@ -265,6 +335,304 @@ class SemanticsController {
+  void scrollRight({FinderBase<SemanticsNode>? finder}) {
+    performAction(finder ?? _find.semantics.scrollable(), SemanticsAction.scrollRight);
+  }
+
+  /// Performs a [SemanticsAction.scrollUp] action on the [SemanticsNode] found
+  /// by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.scrollUp]
+  void scrollUp({FinderBase<SemanticsNode>? finder}) {
+    performAction(finder ?? _find.semantics.scrollable(), SemanticsAction.scrollUp);
+  }
+
+  /// Performs a [SemanticsAction.scrollDown] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.scrollDown]
+  void scrollDown({FinderBase<SemanticsNode>? finder}) {
+    performAction(finder ?? _find.semantics.scrollable(), SemanticsAction.scrollDown);
+  }
+
+  /// Performs a [SemanticsAction.increase] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.increase]
+  void increase(FinderBase<SemanticsNode> finder) {
+    performAction(finder, SemanticsAction.increase);
+  }
+
+  /// Performs a [SemanticsAction.decrease] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.decrease]
+  void decrease(FinderBase<SemanticsNode> finder) {
+    performAction(finder, SemanticsAction.decrease);
+  }
+
+  /// Performs a [SemanticsAction.showOnScreen] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.showOnScreen]
+  void showOnScreen(FinderBase<SemanticsNode> finder) {
+    performAction(
+      finder,
+      SemanticsAction.showOnScreen,
+      checkForAction: false,
+    );
+  }
+
+  /// Performs a [SemanticsAction.moveCursorForwardByCharacter] action on the
+  /// [SemanticsNode] found by `finder`
+  ///
+  /// If `shouldModifySelection` is true, then the cursor will begin or extend
+  /// a selection.
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  ///   [SemanticsAction.moveCursorForwardByCharacter]
+  void moveCursorForwardByCharacter(FinderBase<SemanticsNode> finder, [bool shouldModifySelection = false]) {
```

Comment:
make shouldModifySelection a named parameter to make the callsites look more self explanatory:

```dart
moveCursorForwardByCharacter(myFinder, true)
```

vs.

```dart
moveCursorForwardByCharacter(myFinder, shouldModifySelection: true)
```

(If we hadn't so many legacy offenders, we would enable a lint that enforces this for boolean parameters)

Reply:
Same for the other cases of this below.